wdl-lint/src/rules/deprecated_placeholder_option.rs

```diff
@@ -0,0 +1,117 @@
+//! A lint rule for flagging placeholders as deprecated.
+
+use wdl_ast::span_of;
+use wdl_ast::v1::Placeholder;
+use wdl_ast::v1::PlaceholderOption;
+use wdl_ast::Diagnostic;
+use wdl_ast::Diagnostics;
+use wdl_ast::Document;
+use wdl_ast::Span;
+use wdl_ast::VisitReason;
+use wdl_ast::Visitor;
+
+use crate::Rule;
+use crate::Tag;
+use crate::TagSet;
+
+/// The identifier for the deprecated placeholder option rule.
+const ID: &str = "DeprecatedPlaceholderOption";
+
+/// Creates a diagnostic for the use of a deprecated `default` placeholder
+/// option.
+fn deprecated_default_placeholder_option(span: Span) -> Diagnostic {
+    Diagnostic::warning(String::from(
+        "use of a deprecated `default` placeholder option",
+    ))
+    .with_rule(ID)
+    .with_highlight(span)
+    .with_fix(
+        "replace the `default` placeholder option with a call to the `select_first()` standard \
+         library function",
+    )
+}
+
+/// Creates a diagnostic for the use of a deprecated `sep` placeholder option.
+fn deprecated_sep_placeholder_option(span: Span) -> Diagnostic {
+    Diagnostic::warning(String::from("use of a deprecated `sep` placeholder option"))
+        .with_rule(ID)
+        .with_highlight(span)
+        .with_fix(
+            "replace the `sep` placeholder option with a call to the `sep()` standard library \
+             function",
```

Comment:
I haven't checked with a Cromwell run, but at least according to the [`v1.0` doc](https://github.com/openwdl/wdl/blob/main/versions/1.0/SPEC.md#standard-library) there is no `sep()` function in the standard library. Which is to say this "fix" might not work for Cromwell folks stuck in `v1.0`.

However I think it makes sense to warn about the deprecation of `default` and `true/false` placeholders in 1.0 documents as they have viable alternatives.

Can we prevent this specific diagnostic in 1.0 docs, while permitting the others?

Reply:
Good catch! I can do this, but it's going to require my other branch to be merged in first where I make the version parsing much easier for lint rules.